src/main/java/org/opensearch/ad/task/ADBatchTaskCache.java

```diff
@@ -70,14 +74,20 @@ protected ADBatchTaskCache(ADTask adTask) {
         this.taskId = adTask.getTaskId();
 
         AnomalyDetector detector = adTask.getDetector();
+        int dimensions = detector.getShingleSize() * detector.getEnabledFeatureIds().size();
         rcfModel = RandomCutForest
             .builder()
-            .dimensions(detector.getShingleSize() * detector.getEnabledFeatureIds().size())
+            .dimensions(dimensions)
             .numberOfTrees(NUM_TREES)
-            .lambda(TIME_DECAY)
+            .timeDecay(TIME_DECAY)
             .sampleSize(NUM_SAMPLES_PER_TREE)
             .outputAfter(NUM_MIN_SAMPLES)
             .parallelExecutionEnabled(false)
+            .compact(true)
+            .precision(Precision.FLOAT_32)
+            .boundingBoxCacheFraction(AnomalyDetectorSettings.BATCH_BOUNDING_BOX_CACHE_RATIO)
+            // same with dimension for opportunistic memory saving
+            .shingleSize(dimensions)
```

Comment:
Why not use `detector.getShingleSize()` as RCF's shingle size? If enabled feature size is 5, and detector single size is 8, then the RCF's shingle size will be 40?

Reply:
Seems `shingleSize` is new parameter of RCF, what's the difference with `dimension`? I don't see `EntityColdStarter.java` set `shingleSize` when create RCF.

Reply:
rcf's shingle size is different from our detector's shingle size.  " If enabled feature size is 5, and detector single size is 8, then the RCF's shingle size will be 40" that's correct.

Reply:
We do use shingleSize at line 344: https://github.com/opensearch-project/anomaly-detection/pull/143/files#diff-235791084d307db4d86529d4f484ef3e6f213af48907cc003298b8e0e2211192R344

Reply:
I explained the shingleSize parameter in the overview: "Enable shingleSize optimization. This optimization will try to save space opportunistically. Usually, we use it as shingleSize(dimension). When a new point comes in, we will look at the point store if there is any overlapping. Say the previously-stored vector is x1, x2, x3, x4, now we add x3, x4, x5, x6. RCF will recognize overlapping x3, x4, and only store x5, x6."

Reply:
will add this as a comment as well

Reply:
Wait! For RCF 2.0 we would need to set dimensions = 40 and shinglesize =8. We can set other values of shingle size -- the code will match opportunistically but we may not see any savings. For shingle size 8, we will get a match and only add 5 new values per item

Reply:
so just ".shingleSize(detector.getShingleSize())"

Reply:
right, will change

@ylwu-amzn  shingleSize in rcf and AD are the same.  Sorry for the confusion.  My explanation about the shingleSize parameter still holds

Reply:
Cool. That part looks clear now.

For this 
> Enable shingleSize optimization. This optimization will try to save space opportunistically. Usually, we use it as shingleSize(dimension). When a new point comes in, we will look at the point store if there is any overlapping. Say the previously-stored vector is x1, x2, x3, x4, now we add x3, x4, x5, x6. RCF will recognize overlapping x3, x4, and only store x5, x6.

Can you add it to specific line of code? Still not quite understand which part in this PR does this optimization.

Reply:
I did.  The changed code looks like

```
rcfModel = RandomCutForest
            .builder()
            .dimensions(dimensions)
            .numberOfTrees(numberOfTrees)
            .timeDecay(TIME_DECAY)
            .sampleSize(NUM_SAMPLES_PER_TREE)
            .outputAfter(NUM_MIN_SAMPLES)
            .parallelExecutionEnabled(false)
            .compact(true)
            .precision(Precision.FLOAT_32)
            .boundingBoxCacheFraction(AnomalyDetectorSettings.BATCH_BOUNDING_BOX_CACHE_RATIO)
            // same with dimension for opportunistic memory saving
            // Usually, we use it as shingleSize(dimension). When a new point comes in, we will
            // look at the point store if there is any overlapping. Say the previously-stored
            // vector is x1, x2, x3, x4, now we add x3, x4, x5, x6. RCF will recognize
            // overlapping x3, x4, and only store x5, x6.
            .shingleSize(shingleSize)
            .build();
```

Reply:
Looks good.